src/rust-crypto/verification.ts

```diff
@@ -75,11 +75,15 @@ export class RustVerificationRequest
         const onChange = async (): Promise<void> => {
             const verification: RustSdkCryptoJs.Qr | RustSdkCryptoJs.Sas | undefined = this.inner.getVerification();
 
-            // If we now have a `Verification` where we lacked one before, or we have transitioned from QR to SAS,
-            // wrap the new rust Verification as a js-sdk Verifier.
             if (verification instanceof RustSdkCryptoJs.Sas) {
                 if (this._verifier === undefined || this._verifier instanceof RustQrCodeVerifier) {
+                    // If we now have a `Verification` where we lacked one before, or we have transitioned from QR to SAS,
+                    // wrap the new rust Verification as a js-sdk Verifier.
```

Comment:
There are two branches to this `if` statement (`if (verification instanceof RustSdkCryptoJs.Sas)`, and `else if (verification instanceof RustSdkCryptoJs.Qr && this._verifier === undefined)`), and this comment applies to both of them. By moving it in here, it's somewhat less clear.

Reply:
Moved the comment back and modified it